### PR TITLE
Remove JAVA_OPTS deprecated in JDK15.

### PR DESCRIPTION
**Pull request by juju2112**
The Concurrent Mark Sweep (CMS) Garbage Collector has been removed in JDK 15 (released Sept. 2020). Graylog won't start at all if any of these JVM_OPTS are included:

`-XX:+UseConcMarkSweepGC -XX:+CMSConcurrentMTEnabled -XX:+CMSClassUnloadingEnabled`

Some references on this:

https://jaxenter.com/cms-gc-jdk-9-java-161723.html

https://openjdk.java.net/jeps/363

## Description
This removes the CMS GC Java opts from DEFAULT_JAVA_OPTS. 

## Motivation and Context
Reported here: Graylog2/graylog-ansible-role#166

## How Has This Been Tested?
I spun up an Ubuntu 20.04 VM, installed JDK15, and installed Graylog via the Ansible role. I then edited the OPTs in vi and witnessed Graylog not starting. 

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: bin/graylogctl [hunk 1]
```diff
@@ -51,7 +51,7 @@ GRAYLOG_CONF=${GRAYLOG_CONF:=/etc/graylog/server/server.conf}
 GRAYLOG_PID=${GRAYLOG_PID:=/tmp/graylog.pid}
 LOG_FILE=${LOG_FILE:=log/graylog-server.log}
 LOG4J=${LOG4J:=}
-DEFAULT_JAVA_OPTS="-Djdk.tls.acknowledgeCloseNotify=true -Xms1g -Xmx1g -XX:NewRatio=1 -server -XX:+ResizeTLAB -XX:+UseConcMarkSweepGC -XX:+CMSConcurrentMTEnabled -XX:+CMSClassUnloadingEnabled -XX:-OmitStackTraceInFastThrow"
+DEFAULT_JAVA_OPTS="-Djdk.tls.acknowledgeCloseNotify=true -Xms1g -Xmx1g -XX:NewRatio=1 -server -XX:+ResizeTLAB -XX:-OmitStackTraceInFastThrow"
```
Comment (bernd) [inline bin/graylogctl:54]: We cannot just remove the flags for all JVM versions. That would most probably the default GC for some JVMs. Please check the lines below on how to do a check for the presence of JVM flags so we can keep the flags for older JVMs.

**Review by bernd — CHANGES_REQUESTED** [review]

File: bin/graylogctl (hunk 1 above)
Comment (juju2112) [reply bin/graylogctl:54]: I was wondering about that. I didn't notice the code below it that checks if the flags are available. I'll see if I can add something like that.

Comment (juju2112) [reply bin/graylogctl:54]: Okay, I added the changes you suggested. I tested them in an Ubuntu 20.04 VM using Java 8 and Java 15. On Java 8, it includes the CMS opts. On Java 15, it doesn't include them. Graylog starts successfully in both instances.

File: bin/graylogctl [hunk 2]
```diff
@@ -51,10 +51,19 @@ GRAYLOG_CONF=${GRAYLOG_CONF:=/etc/graylog/server/server.conf}
...
 if $JAVA_CMD -XX:+PrintFlagsFinal 2>&1 |grep -q UseParNewGC; then
 	DEFAULT_JAVA_OPTS="${DEFAULT_JAVA_OPTS} -XX:+UseParNewGC"
 fi
+if $JAVA_CMD -XX:+PrintFlagsFinal 2>&1 |grep -q UseConcMarkSweepGC; then
+	DEFAULT_JAVA_OPTS="${DEFAULT_JAVA_OPTS} -XX:+UseConcMarkSweepGC"
+fi
+if $JAVA_CMD -XX:+PrintFlagsFinal 2>&1 |grep -q CMSConcurrentMTEnabled; then
+	DEFAULT_JAVA_OPTS="${DEFAULT_JAVA_OPTS} -XX:+CMSConcurrentMTEnabled"
```
Comment (mpfz0r) [inline bin/graylogctl:62]: Hmm, I guess these CMS ones are all related to `UseConMarkSweepGC`.
I'd simplify this and add all CMS options within the first check

Comment (juju2112) [reply bin/graylogctl]: I thought about doing that, but I wasn't sure. I like simpler.

**Review by mpfz0r — APPROVED** [review]
LGTM

**Comment by mpfz0r** [comment]
@bernd can you approve? merging is blocked. build failure is from a frontend test :shrug:

**Review by bernd — APPROVED** [review]

**Pull request merged**